spec/reservation_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe Reservation, type: :model do
+  before(:all) do
+    @user = User.create(username: "john doe1#{SecureRandom.hex}", password: '123456', password_confirmation: '123456')
```

Comment:
Great job here, but if you use the `create` method it will try to save the record in the database even if it's not valid, and you will not have a chance to verify the validity. Here I would suggest using `User.new` instead, to be able to check the validity using `.valid?`